habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -386,6 +386,290 @@ def step(self, *args, is_last_action, **kwargs):
+    """
+    The robot base motion is constrained to the NavMesh and controlled with velocity commands integrated with the VelocityControl interface.
+    Optionally cull states with active collisions if config parameter `allow_dyn_slide` is True
+    The robot is given a target waypoint and output a velocity command
+    """
+
+    def __init__(self, *args, config, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
+        self.base_vel_ctrl.controlling_lin_vel = True
+        self.base_vel_ctrl.lin_vel_is_local = True
+        self.base_vel_ctrl.controlling_ang_vel = True
+        self.base_vel_ctrl.ang_vel_is_local = True
+        self.controller = Controller(self._config.lin_speed, self._config.ang_speed)
+
+    @property
+    def action_space(self):
+        lim = 20
+        return spaces.Dict(
+            {
+                self._action_arg_prefix
+                + "base_vel": spaces.Box(
+                    shape=(2,), low=-lim, high=lim, dtype=np.float32
+                )
+            }
+        )
+
+    def _capture_robot_state(self):
+        return {
+            "forces": self.cur_robot.sim_obj.joint_forces,
+            "vel": self.cur_robot.sim_obj.joint_velocities,
+            "pos": self.cur_robot.sim_obj.joint_positions,
+        }
+
+    def _set_robot_state(self, set_dat):
+        """"
+        Keep track of robot's basic info
+        """
+        self.cur_robot.sim_obj.joint_positions = set_dat["forces"]
+        self.cur_robot.sim_obj.joint_velocities = set_dat["vel"]
+        self.cur_robot.sim_obj.joint_forces = set_dat["pos"]
+
+    def update_base(self):
+        """
+        Update the robot base
+        """
+
+        ctrl_freq = self._sim.ctrl_freq
+
+        before_trans_state = self._capture_robot_state()
+
+        trans = self.cur_robot.sim_obj.transformation
+        rigid_state = habitat_sim.RigidState(
+            mn.Quaternion.from_matrix(trans.rotation()), trans.translation
+        )
+
+        target_rigid_state = self.base_vel_ctrl.integrate_transform(
+            1 / ctrl_freq, rigid_state
+        )
+        end_pos = self._sim.step_filter(
+            rigid_state.translation, target_rigid_state.translation
+        )
+
+        target_trans = mn.Matrix4.from_(
+            target_rigid_state.rotation.to_matrix(), end_pos
+        )
+        self.cur_robot.sim_obj.transformation = target_trans
+
+        if not self._config.get("allow_dyn_slide", True):
+            # Check if in the new robot state the arm collides with anything.
+            # If so we have to revert back to the previous transform
+            self._sim.internal_step(-1)
+            colls = self._sim.get_collisions()
+            did_coll, _ = rearrange_collision(
+                colls, self._sim.snapped_obj_id, False
+            )
+            if did_coll:
```

Comment:
did collide?
collisions?
better names helpful here maybe

Reply:
I just reused the naming conversion in Habitat. Thank you for raising this!